src/main/java/xbdd/webapp/resource/feature/Feature.java

```diff
@@ -201,6 +196,165 @@ public DBObject putFeature(@BeanParam final Coordinates coordinates, @PathParam(
 		}
 	}
 
+	@PUT
+	@Path("/step/{product}/{major}.{minor}.{servicePack}/{build}/{featureId:.+}")
+	@Consumes("application/json")
+	public Response updateStepWithPatch(@BeanParam final Coordinates coordinates, @PathParam("featureId") final String featureId,
+								   @Context final HttpServletRequest req, final DBObject patch) {
+		try {
+			final DB db = this.client.getDB("bdd");
+			final DBCollection collection = db.getCollection("features");
+			final BasicDBObject example = coordinates.getReportCoordinatesQueryObject().append("id", featureId);
+			final BasicDBObject storedFeature = (BasicDBObject) collection.findOne(example);
+
+			final int stepLine = (int) patch.get("line");
+			final String status = (String) patch.get("status");
+			final String scenarioId = (String) patch.get("id");
+
+			final BasicDBObject featureToUpdate = (BasicDBObject) storedFeature.copy();
+			final BasicDBObject scenarioToUpdate = getScenarioById(scenarioId, featureToUpdate);
+
+
+			boolean found = false;
+
+			if(scenarioToUpdate.get("background") != null) {
+				final BasicDBObject backgroundToUpdate = (BasicDBObject) (scenarioToUpdate.get("background"));
+				final BasicDBList backgroundStepsToUpdate = (BasicDBList) (backgroundToUpdate.get("steps"));
+				found = updateSteps(backgroundStepsToUpdate, stepLine, status);
+			}
+			if(!found) {
+				final BasicDBList stepsToUpdate = (BasicDBList) (scenarioToUpdate.get("steps"));
+				updateSteps(stepsToUpdate, stepLine, status);
+			}
+			featureToUpdate.put("statusLastEditedBy", req.getRemoteUser());
+			featureToUpdate.put("lastEditOn", new Date());
+			featureToUpdate.put("calculatedStatus", calculateStatusForFeature(featureToUpdate));
+			collection.save(featureToUpdate);
+			Feature.embedTestingTips(featureToUpdate, coordinates, db); // rembed testing tips.
```

Comment:
Whats this comment for?

Reply:
Old code forgot to delete, will implement embedTestingTips later